pkg/logql/shardmapper_test.go

```diff
@@ -675,7 +675,11 @@ func TestMapping(t *testing.T) {
 		{
 			in: `1 + sum by (cluster) (rate({foo="bar"}[5m]))`,
 			expr: &BinOpExpr{
-				op:         OpTypeAdd,
+				op: OpTypeAdd,
```

Comment:
We need example with this new operator. The binary operation should never be shardable so I think we're fine.

Reply:
thanks @cyriltovena for your comments, make sense, added example for new operator in shardmapper unit test - https://github.com/grafana/loki/pull/4391/commits/9cb1a8142eee69112483ee5178b3aade77669bd0

Reply:
It's currrently implemented like this:
```
func (e *BinOpExpr) Shardable() bool {
	return shardableOps[e.op] && e.SampleExpr.Shardable() && e.RHS.Shardable()
}
```

However, I think we should prohibit sharding when we're changing the label groupings such as in `on` or `without` as it effectively changes the labels and thus invalidates correctness guarantees in sharded aggregation. @garrettlish can you make the change to the `Shardable` method to return `false` if either `on` or `without` are specified?